e2e/test/provisioning/ProvisioningServiceClientE2ETests.IndividualEnrollmentsClient.cs

```diff
@@ -0,0 +1,392 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Net;
+using System.Security.Cryptography.X509Certificates;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure;
+using FluentAssertions;
+using FluentAssertions.Specialized;
+using Microsoft.Azure.Devices.E2ETests.Helpers;
+using Microsoft.Azure.Devices.Provisioning.Service;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+namespace Microsoft.Azure.Devices.E2ETests.Provisioning
+{
+    public partial class ProvisioningServiceClientE2ETests : E2EMsTestBase
```

Comment:
Why partial classes?

Reply:
I mean, why not just give each their own class with a name that is appropriate, e.g. IndividualEnrollmentsClientsTests.

Reply:
The test file was getting too large to include test cases for both individual enrollments and enrollment groups. I could have split them into separate files, but I wanted to make use of some common methods for both. So, I decided to use partial classes instead.

Reply:
I don't see a good argument for partial classes. I agree we don't want files with too many tests and in splitting them up.

I see partial classes as a solution to a specific set of problems, not something to be used because it feels convenient. They come at a conceptual cost for code readers. Why is this class partial? Where is the other implementation? Navigating to definition is harder, etc.

In this case, can we just make them different class names?